### PR TITLE
fix DPLL state comparison

### PR DESCRIPTION
**Pull request by jzding**

### CODE REVIEW

**Comment by nishant-parekh** [comment]
Just 1 small comment otherwise cool way to implement this :)

File: pkg/dpll/dpll.go [hunk 1]
```diff
@@ -197,9 +196,26 @@ func (d *DpllConfig) MonitorDpll(processCfg config.ProcessConfig) {
...
 }
 
-func (d *DpllConfig) getState(pstate, fstate int64) int64 {
-	//TODO: Fix the logic to get correct lowest state
-	return int64(math.Min(float64(pstate), float64(fstate)))
+// getStateQuality maps the state with relatively worse signal quality with
+// a lower number for easy comparison
+func (d *DpllConfig) getStateQuality() map[int64]float64 {
+	return map[int64]float64{
+		DPLL_UNKNOW:        -1,
```
Comment (josephdrichard) [inline pkg/dpll/dpll.go:203]: DPLL_UNKNOWN?

File: pkg/dpll/dpll.go [hunk 2]
```diff
@@ -197,9 +196,26 @@ func (d *DpllConfig) MonitorDpll(processCfg config.ProcessConfig) {
...
+		DPLL_INVALID:       0,
+		DPLL_FREERUN:       1,
+		DPLL_HOLOVER:       2,
```
Comment (josephdrichard) [inline pkg/dpll/dpll.go:206]: DPLL_HOLDOVER?

File: pkg/dpll/dpll.go [hunk 3]
```diff
@@ -197,9 +196,26 @@ func (d *DpllConfig) MonitorDpll(processCfg config.ProcessConfig) {
...
+		DPLL_LOCKED_HO_ACQ: 3,
+		DPLL_LOCKED:        4,
```
[the remark this answers is not in the document]
Comment (jzding) [reply pkg/dpll/dpll.go]: I think you are right. according to  ITU-T G.781 section 6.3.1 Auto selection operation.
```
– Locked mode, acquiring holdover: This is a temporary mode when coming from freerun, to
acquire holdover memory. It behaves as specified below.
 As shown on Figure 23, some time is required before the holdover memory is acquired.
– Locked mode, holdover acquired: This is a steady state mode, entered when holdover
memory is acquired. It behaves as specified below. 
```

**Review by nishant-parekh — APPROVED** [review]

**Review by josephdrichard — APPROVED** [review]
/lgtm

**Pull request merged**